src/Shared/CertificateGeneration/WindowsCertificateManager.cs

```diff
@@ -28,11 +28,20 @@ protected override bool IsExportable(X509Certificate2 c)
             // For the first run experience we don't need to know if the certificate can be exported.
             return true;
 #else
-            using var key = c.GetRSAPrivateKey();
-            return (key is RSACryptoServiceProvider rsaPrivateKey &&
-                    rsaPrivateKey.CspKeyContainerInfo.Exportable) ||
-                (key is RSACng cngPrivateKey &&
-                    cngPrivateKey.Key.ExportPolicy == CngExportPolicies.AllowExport);
+            try
+            {
+                using var key = c.GetRSAPrivateKey();
+                return (key is RSACryptoServiceProvider rsaPrivateKey &&
+                        rsaPrivateKey.CspKeyContainerInfo.Exportable) ||
+                       (key is RSACng cngPrivateKey &&
+                        cngPrivateKey.Key.ExportPolicy == CngExportPolicies.AllowExport);
+            }
+            catch (CryptographicException)
+            {
+                // Although undocumented, GetRSAPrivateKey may, under unidentified circumstances,
+                // throw a "Keyset does not exist" CryptographicException
```

Comment:
We don't think we want to do this.

This I believe could result in the creation of multiple certificates if there is a persistent issue that is causing an exception here.

We'd rather see the tool fail in these cases and get a chance to understand what's going on wrong.